core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -148,6 +148,20 @@ case object TaskKilled extends TaskFailedReason {
 
 /**
  * :: DeveloperApi ::
+ * Task requested the driver to commit, but was denied.
+ */
+@DeveloperApi
+case class TaskCommitDenied(
+    jobID: Int,
+    splitID: Int,
```

Comment:
Since we've renamed "split" to "partition", this should either be partitionID or taskID.